sympy/stats/drv_types.py

```diff
@@ -28,13 +29,68 @@ def check(lamda):
     def pdf(self, k):
         return self.lamda**k / factorial(k) * exp(-self.lamda)
 
-    def sample(self):
-        u = random.uniform(0, 1)
-        inf = self.set.inf
-        while u > self.cdf(inf):
-            inf += 1
-        return inf
+    def _cdf(self, k):
```

Comment:
This does not yet have any effect, because this method is not called. Look at SingleContinuousDistribution class in crv.py, specifically its _cdf and cdf methods. Do the same in SingleDiscreteDistribution in drv.py

Reply:
And add a test for the new CDF when it works. The tests are in `stats/tests/test_discrete_rv.py`